esmvaltool/_recipe.py

```diff
@@ -744,16 +739,18 @@ def _get_preprocessor_task(variables, profiles, config_user, task_name):
                 variable['preprocessor'], variable['short_name'])
     variables = _limit_datasets(variables, profile,
                                 config_user.get('max_datasets'))
-
+    for variable in variables:
+        _add_cmor_info(variable)
     # Create preprocessor task(s)
     derive_tasks = []
     if variable.get('derive'):
         # Create tasks to prepare the input data for the derive step
         derive_profile, profile = _split_derive_profile(profile)
-
         derive_input = _get_derive_input_variables(variables, config_user)
 
         for derive_variables in derive_input.values():
+            for derive_variable in derive_variables:
+                _add_cmor_info(derive_variable)
```

Comment:
I think you need
```suggestion
                _add_cmor_info(derive_variable, override=True)
```
otherwise it will not update the entries that are already present with the correct value